src/gameboy/main.cpp

```diff
@@ -85,8 +85,20 @@ int main(int argc, char *argv[]) {
 	}
 
 	auto cartridge = std::make_unique<Cartridge>(rom_path);
+
+	/**
+	 *  Displaying the Meta Information of the Cartridge if it was a successful
+	 * load, else the TVP freezes
+	 */
+	if (cartridge->meta_data->is_logo_valid) {
+		cartridge->meta_data->display_meta();
+	} else {
+		cout << "Failed to fetch ROM Details" << endl;
+	}
+
 	auto controller = std::make_unique<Controller>();
-	auto video = make_unique<Video>(controller.get());
+	auto video =
+	    make_unique<Video>(controller.get(), cartridge->meta_data->title);
```

Comment:
Pass the entire metadata object to video, and let it use only the name. If the name is empty, let it set it to "Welcome to TVP"